cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/requestrate/RequestRateModule.java

```diff
@@ -14,18 +26,110 @@
  */
 public class RequestRateModule extends Module {
 
+    private static final Logger logger = LoggerFactory.getLogger(RequestRateModule.class);
+
+    private static final String PERIOD_PROP = "period";
+
+    private static final String DEFAULT_PERIOD = "1";
+
+    private static final String TIMEUNIT_PROP = "timeunit";
+
+    private static final String DEFAULT_TIMEUNIT = "SECONDS";
+
+    private static final String UPDATE_SUFFIX = "_update";
+
+    private static final String SELECT_SUFFIX = "_select";
+
+    private final MetricRegistry metricsRegistry = new MetricRegistry();
+
+    private final Meter updateRequests;
+
+    private final Meter selectRequests;
+
+    private final String service;
+
+    private final String updateService;
+
+    private final String selectService;
+
+    private final int period;
+
+    private final TimeUnit timeunit;
+
+    private final long rateFactor;
+
+    private final Timer timer;
+
     /**
      * Constructor.
      *
      * @param configuration Module configuration
      * @param reporters     Reporter list
+     * @throws ConfigurationException in case the provided module configuration is not valid
      */
-    public RequestRateModule(ModuleConfiguration configuration, List<Reporter> reporters) {
+    public RequestRateModule(ModuleConfiguration configuration, List<Reporter> reporters)
+            throws ConfigurationException {
         super(configuration, reporters);
+
+        RequestRateConfiguration config = RequestRateConfiguration.create(configuration.options);
+        service = configuration.measurement;
+        period = config.period();
+        timeunit = config.timeunit();
+        rateFactor = timeunit.toSeconds(1);
+
+        logger.info("RequestRate module initialized with {} period and {} timeunit.", period, timeunit.name());
+        updateService = service + UPDATE_SUFFIX;
+        selectService = service + SELECT_SUFFIX;
+        updateRequests = metricsRegistry.meter(updateService);
+        selectRequests = metricsRegistry.meter(selectService);
+        timer = new Timer();
+        timer.schedule(new RequestRateTask(), timeunit.toMillis(period));
+    }
+
+    @Override
+    protected boolean isForReporting(Query query) {
+        return false;
     }
 
     @Override
     public Measurement transform(Query query) {
+        // Future work: Separate marks for request types
```

Comment:
This future work comment, is it obsolete since I see below that you separate select and update?